### PR TITLE
Deprecate (remove) generate-id

### PR DESCRIPTION
**Pull request by tmorehouse**
Removed need for generate ID mixin.

Users must supply an ID on form input (file, radio,input, etc) components if they want ARIA compliance. IF they do not supply ID(s) then many of the aria attributes will not appear (but wont break functionality)

Allow users to override the `b-form-fieldset` `label`'s attribute `for` (needed in special cases like groups of checkboxes). If no `for` prop is supplied looks for first input to associated to.  For radio buttons, it points to the group of radios (`role="radiogroup"`) rather than the first radio button

Fixed bug where `form-file` with `plain` set to `true` wouldn't display the file input.

Fixed bug where MS Edge browser pukes on `<input type="file">` inside `<label>` (issue #315). Fixed by changing `label` to `div`

### CODE REVIEW

**Comment by tmorehouse** [comment]
After looking at `form-radio` code, I wonder if a `form-checkbox-group` should be created (which takes an array/object of options)?

**Comment by tmorehouse** [comment]
After this merge, I think we can safely delete `generate-id.js` mixin.

**Comment by SirLamer** [comment]
I was requested to review, but as I have had nothing to do with the form elements to date, I will abstain unless called back explicitly.

**Comment by tmorehouse** [comment]
@SirLamer I added you just because of our original discussion on using auto-ids (but unfortunately Vue SSR foobars it up for us :( ).

**Comment by SirLamer** [comment]
Ohhh okay, well @pi0 said he doesn't want auto-IDs so I kinda forgot about that. ;-)

I think auto-assigning a unique ID for accessibility functionality in the case where an ID is not provided is a good idea, I do that with my own form elements I use in my application.

**Comment by tmorehouse** [comment]
Went through a few more component and did a bit of cleanup.

File: lib/mixins/generate-id.js [hunk 1]
```diff
@@ -4,12 +4,7 @@
  * it is better to prefer references/refs and events over relying on generated id.
  * @see https://github.com/bootstrap-vue/bootstrap-vue/issues/281
  *
- * Currently is being used in:
- *  - dropdown.vue
- *  - form-checkbox-vue
- *  - form-file.vue
- *  - form-input.vue
- *  - form-select.vue
+ * No longer used
```
Comment (pi0) [inline lib/mixins/generate-id.js]: So is it is safe to just remove this mixin file ? :D

Comment (tmorehouse) [inline lib/mixins/generate-id.js]: I think it is.  I went through most of the components to see if it was referenced anywhere else and didn't s it.

**Pull request merged**